pkg/util/kube.go

```diff
@@ -18,3 +35,139 @@ const (
 	// Kube annotation for podman volume image.
 	VolumeImageAnnotation = "volume.podman.io/image"
 )
+
+// move it here ?
```

Comment:
> but you should probably keep it under `domain/infra/abi/kube.go`.

I am not sure to follow, as the content mention here cannot built for the remote target given my understanding (might be wrong, as I am not familiar with go).

https://github.com/containers/podman/blob/a88581f38fed84d6959ddedf8fbd823e50f79e2a/pkg/domain/infra/abi/play.go#L1

Moreover, would it make sense to import functions in `domain/infra/abi/kube.go` from `cmd/podman/kube/` ?

> If possible we should keep the remote/local build closer considered that the logic is very similar.

Sorry, I am not sure to understand what you are expecting here. 

Before putting the parsing logic under `pkg/util/kube.go` I tried to add it inside `cmd/podman/kube/`.  However I faced a circular issue when I was trying to import functions in `cmd/podman/kube/` from `domain/infra/abi/kube.go`.

I used the `pkg/util/kube.go` to share the code between the remote doing the build and the native doing the build, to keep a single source of truth for the parsing logic.

> Also it would be great if you could add a podman machine e2e test (see /home/mariolet/github/podman/pkg/machine/e2e/README.md).

I tried to write one, I will try to add more when I will have some time available.

> Anyway thank you for this PR @fixomatic-ctrl, this is a great feature to add.

Thanks you for your time and comments!

Reply:
Image builds happen server side. A remote client cannot build an image without a running podman server.

That said there are 2 possible approaches:
1) as for `podman build`, `podman kube play` sends the build context to the server. So that the podman server has access to the Dockerfile and to the files required by the build. 
2) `podman kube play` runs `podman build` as a preparation step (something like what you have done in this PR)

I tend to prefer the first approach as it seams more consistent with how Podman works today.

Reply:
I must admit I have some doubt about it being fully on the server side. I read the discussion on the ticket ( https://github.com/containers/podman/issues/14527) related to sending the all context directory to the server as a tar, just like this is done for the podman build command, but some people commented the following 

> Sending a (possible) large directory by default is very bad.[^1]

[^1]: https://github.com/containers/podman/issues/14527#issuecomment-2180869202

I do not have a problem sending the full context, and I would be willing to try making the change. 

It seems that [according to the documentation](https://docs.podman.io/en/latest/_static/api.html#tag/containers/operation/PlayKubeLibpod), the libpod endpoint already have some kind of support for a tar 

![image](https://github.com/user-attachments/assets/14031bf9-2f51-47bd-b58b-8beee80cb8d0)